### PR TITLE
[Angular 17] Control flow syntax for 'switch'

### PR DESCRIPTION
**Pull request by qmonmert**
Following #24316 

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by mraible — APPROVED** [review]

File: generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs [hunk 1]
```diff
@@ -26,25 +26,31 @@
...
-    <div [ngSwitch]="account !== null">
-      <div class="alert alert-success" *ngSwitchCase="true">
-      @if (account) {
-        <span id="home-logged-message" <%= jhiPrefix %>Translate="home.logged.message"
-          [translateValues]="{ username: account.login }">__jhiTransformTranslate__('home.logged.message', { "username": "{{ account.login }}" })</span>
-      }
-      </div>
+    <div>
+      @switch (account !== null) {
+        @case (true) {
```
Comment (mshima) [inline generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs:31]:
```suggestion
      @if (account !== null) {
```

File: generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs [hunk 2]
```diff
@@ -26,25 +26,31 @@
...
+          <div class="alert alert-success">
+          @if (account) {
+            <span id="home-logged-message" <%= jhiPrefix %>Translate="home.logged.message"
+              [translateValues]="{ username: account.login }">__jhiTransformTranslate__('home.logged.message', { "username": "{{ account.login }}" })</span>
+          }
+          </div>
+        }
```
Comment (mshima) [inline generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs:38]:
```suggestion
```

File: generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs [hunk 3]
```diff
@@ -26,25 +26,31 @@
...
 
-      <div class="alert alert-warning" *ngSwitchCase="false">
-        <span <%= jhiPrefix %>Translate="global.messages.info.authenticated.prefix">__jhiTransformTranslate__('global.messages.info.authenticated.prefix')</span>
-        <a class="alert-link" (click)="login()" <%= jhiPrefix %>Translate="global.messages.info.authenticated.link">__jhiTransformTranslate__('global.messages.info.authenticated.link')</a><span <%= jhiPrefix %>Translate="global.messages.info.authenticated.suffix">__jhiTransformTranslate__('global.messages.info.authenticated.suffix')</span>
-      </div>
-      <%_ if (generateUserManagement) { _%>
+        @case (false) {
```
Comment (mshima) [inline generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs:40]:
```suggestion
        } @else {
```

Comment (mshima) [inline generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs:40]:
```suggestion
        } @else {
```

File: generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs [hunk 4]
```diff
@@ -26,25 +26,31 @@
...
-        <span <%= jhiPrefix %>Translate="global.messages.info.register.noaccount">__jhiTransformTranslate__('global.messages.info.register.noaccount')</span>&nbsp;
-        <a class="alert-link" routerLink="/account/register" <%= jhiPrefix %>Translate="global.messages.info.register.link">__jhiTransformTranslate__('global.messages.info.register.link')</a>
-      </div>
-      <%_ } _%>
+          <div class="alert alert-warning">
+            <span <%= jhiPrefix %>Translate="global.messages.info.register.noaccount">__jhiTransformTranslate__('global.messages.info.register.noaccount')</span>&nbsp;
+            <a class="alert-link" routerLink="/account/register" <%= jhiPrefix %>Translate="global.messages.info.register.link">__jhiTransformTranslate__('global.messages.info.register.link')</a>
+          </div>
+        <%_ } _%>
+        }
```
Comment (mshima) [inline generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs:52]:
```suggestion
```

**Review by mshima — COMMENTED** [review]
Can we use if/else?

File: generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs (hunk 4 above)
Comment (qmonmert) [reply generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs]: yes good idea, I updated

File: generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs [hunk 5]
```diff
@@ -26,25 +26,27 @@
...
-      }
-      </div>
-
-      <div class="alert alert-warning" *ngSwitchCase="false">
-        <span <%= jhiPrefix %>Translate="global.messages.info.authenticated.prefix">__jhiTransformTranslate__('global.messages.info.authenticated.prefix')</span>
-        <a class="alert-link" (click)="login()" <%= jhiPrefix %>Translate="global.messages.info.authenticated.link">__jhiTransformTranslate__('global.messages.info.authenticated.link')</a><span <%= jhiPrefix %>Translate="global.messages.info.authenticated.suffix">__jhiTransformTranslate__('global.messages.info.authenticated.suffix')</span>
-      </div>
-      <%_ if (generateUserManagement) { _%>
+    <div>
+      @if (account !== null) {
```
Comment (mraible) [inline generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs:30]: I tried to keep the same HTML as before. In this case, I think the `if` should be before the `<div>`.

Comment (mshima) [reply generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs:30]: The original code is a ngSwitch, the div is written for true or false.
This change would be organizational.

Comment (qmonmert) [reply generators/angular/templates/src/main/webapp/app/home/home.component.html.ejs:30]: with the `ngSwitch` this `div` was always present

**Review by mraible — DISMISSED** [review]

File: generators/angular/templates/src/main/webapp/app/layouts/navbar/navbar.component.html.ejs [hunk 6]
```diff
@@ -212,48 +213,54 @@
         }
         </a>
         <ul class="dropdown-menu" ngbDropdownMenu aria-labelledby="account-menu">
+          @switch (account !== null) {
```
Comment (mraible) [inline generators/angular/templates/src/main/webapp/app/layouts/navbar/navbar.component.html.ejs:216]: Should this be an `@if/@else` as well?

Comment (qmonmert) [reply generators/angular/templates/src/main/webapp/app/layouts/navbar/navbar.component.html.ejs]: yes I updated :)

**Review by mshima — APPROVED** [review]

**Pull request merged**